lib/event_sourceable.rb

```diff
@@ -3,4 +3,15 @@
 require "event_sourceable/event_store"
 
 module EventSourceable
+  @metadata = {}
```

Comment:
Based on discussion with @mpeychich this is not thread safe and probably won't work as implemented.

Consider using thread local store which I'm not familiar with.  Found this on the topic: https://www.rubytapas.com/2016/11/20/ruby-thread-local-variables/